File: web2-project/src/app/model/device.ts
import { Address } from "./address";

export enum DeviceType{
	Prekidac,
	Osigurac,
	Transformator,
	Diskonektor
}

export class Device {
	
	public type:DeviceType;
	public id:number;
	public name:string;
	public address:Address;
	public longCoord:number;
	public latCoord:number;
	
	constructor(type:DeviceType, name:string, addr:Address, longcoord:number, latcoord:number)
	{
		this.type = type;
		this.name = name;
		this.address = addr;
		this.longCoord = longcoord;
		this.latCoord = latcoord;
	}
	
}

File: Web2BackEnd/Web2BackEnd/Models/Device.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Web2BackEnd.Models
{
    public class Device : GenericModel
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }
        [Required]
        public DeviceType Type { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public Address Address { get; set; }
        [Required]
        public float LongCoord { get; set; }
        [Required]
        public float LatCoord { get; set; }

    }

    public enum DeviceType
    {
        Prekidac,
        Osigurac,
        Transformator,
        Diskonektor
    }
}
